src/common/time.ts

```diff
@@ -0,0 +1,33 @@
+import dayjs from 'dayjs';
+import utc from 'dayjs/plugin/utc';
+
+dayjs.extend(utc);
+
+export type TimeUnit =
+  | 'year'
+  | 'month'
+  | 'week'
+  | 'day'
+  | 'hour'
+  | 'minute'
+  | 'second';
+
+export class Time {
+  public static now(initialDate?: string | Date | number) {
+    return dayjs(initialDate).utc().toDate();
+  }
+
+  public static dateWithTimeAdded(
+    amount: number,
+    unit: TimeUnit,
+    initialDate?: Date,
+  ): Date {
+    const date = dayjs(initialDate ?? Time.now()).utc();
+    return date.add(amount, unit).toDate();
+  }
+
+  public static isBefore(date: Date, currentDate?: Date): boolean {
```

Comment:
The `currentDate` is a bit confusing as if you want to check it vs now, someone should pass the param explicitly as the naming of the function does not suggest that a comparison with one is the default behaviour